graphql/resolvers.go

```diff
@@ -495,7 +494,23 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
 		}
 		testPointers = append(testPointers, &apiTest)
 	}
-	return testPointers, nil
+
+	totalTestCount, err := r.sc.GetTestCountByTaskIdAndFilters(taskID, "", []string{}, task.Execution)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error getting total test count: %s", err.Error()))
+	}
```

Comment:
this will be the same value every single time the resolver is called. therefore it should be calculated once. what i did for the patch resolver is add a `taskCount` field which is it's own resolver. that way we avoid making one db call every time this resolver is called.

Reply:
It is possible that the number of testresults can increase throughout the execution of a task, so I don't think its true that the value will be same every time this resolver is called. We always show the filtered test count along with the total test counts and they are read together. By calculating the numbers at the same time, we can ensure the user is seeing the correct representation of testresult counts.

Reply:
I think the best approach would be to keep both of the count fields conditionally resolvable so we only fetch them from the UI when the filters change, not when the page or limit changes.